Sources/socktainer/Clients/ClientImageService.swift

```diff
@@ -27,9 +30,46 @@ struct ClientImageService: ClientImageProtocol {
             // Handle specific error if needed
             throw ClientImageError.notFound(id: id)
         }
-
         try await ClientImage.delete(reference: id, garbageCollect: false)
-
     }
 
+    func pull(image: String, tag: String?, platform: Platform, logger: Logger) async throws -> AsyncThrowingStream<String, Error> {
+        let reference: String
+        if let tag = tag, !tag.isEmpty {
+            reference = "\(image):\(tag)"
+        } else {
+            reference = image
+        }
+        return AsyncThrowingStream { continuation in
+            logger.info("Starting to pull image \(reference) for platform \(platform.description)")
+            continuation.yield("Trying to pull \(reference)")
+            Task {
+                do {
+                    let handler: ProgressUpdateHandler = { events in
+                        for event in events {
+                            // NOTE: (vkhitrin) for the time being, the only valuable human readable
+                            //                  event is the size of the blob
+                            switch event {
+                            case .addSize(let size), .setSize(let size):
+                                let message = "Pulled \(size) bytes"
+                                logger.info("\(message)")
```

Comment:
suggest: move it to debug as well ?

Reply:
I was contemplating about making it a debug since it was the only "valuable" info returned from `ClientImage.pull`, but I assume for most scenarios there is no benefit of printing it both client side and server side.

Will set the severity to debug.

Reply:
yes I think you're more interested in the info from client side. And when debugging the server side but I may be wrong